docs/uberon_build_pipeline.md

```diff
@@ -1,83 +1,77 @@
-           - tmp/update-stages
-             - tmp/developmental-stage-ontologies/src/ssso-merged.obo
-             - tmp/developmental-stage-ontologies/src/mmusdv/mmusdv.obo
-1. Imports
-  - tmp/seed.owl
-    - tmp/cl-core.obo
+## Overview
 
+Uberon used to have a completely custom, ad-hoc build pipeline that only
+Chris Mungall knew how to run. Since 2021, the pipeline was migrated to
+a more standard ODK-based setup, with all the Uberon-specific bits
+handled by the custom `src/ontology/uberon.Makefile`.
 
+The repository now uses the typical ODK layout, and the following
+aspects of the pipeline are managed by the ODK in a standard way:
+
+* the imports;
+* the production of the “variants” of the main release artifacts
+  (`-base`, `-full`, `-simple`, etc.);
+* most of the reports and QC checks.
+
+The non-standard parts of the pipeline concerns mainly:
+
+* the production of the main `uberon.owl` product;
+* the production of ontology subsets;
+* the production of the cross-species [bridge files](bridges.md) and
+  [collected/composite ontologies](combined_multispecies.md);
+* the production of extra QC checks and reports (notably the checks for
+  violations of taxon constraints and all the checks involving the
+  bridge files).
+
+## Notes about the custom pipelines
+
+This section is _not_ intended to provide a complete explanation of how
+the custom pipeline works. Not that such an explanation would not be
+useful, but it would be likely to become quickly out-of-sync with the
+actual pipeline code. Ultimately, the reference for how those pipelines
+work is the `uberon.Makefile` file. Consider this section merely as a
+guide to orient yourself within that Makefile.
+
+Unless otherwise noted:
+
+* all filenames are relative to the `src/ontology` directory;
+* all steps are performed with ROBOT.
+
+### Building the main `uberon.owl` product
+
+Two-steps process:
+
+* `tmp/uberon-edit.owl`: made by merging the main source file
+  (`uberon-edit.obo`) along with the ODK import module and all the
+  various components; OWL macros are expanded in that step;
+* `uberon.owl`: reasoning step.
+
+The second step notably includes a materialisation phase that is
+particularly memory intensive and that must therefore be excluded when
+running the pipeline on memory-constrained machines (any machine with
+less than 32 GB of RAM is considered to be memory-constrained when
+Uberon is concerned). Run the pipeline with `GH_ACTION=true` to exclude
+that materialisation step (do _not_ do that when trying to produce an
+official release!).
+
+
+### Building the multi-species ontologies
+
+This is decomposed in several steps that are mostly isolated from each
+other, so that each step can be run independently.
+
+1. Building Uberon itself (`sh run.sh make uberon.owl`).
+2. Mirroring the taxon-specific ontologies (`sh run.sh make MIR=true
+   IMP=true all_local_imports`).
+3. Mirroring the externally provided mapping sets (`sh run.sh make
+   refresh-mapings`).
```

Comment:
Indeed there is a missing `p`. But I don’t understand what you mean with “at odds with the refresh imports series of commands? Mapping not”.

We have `refresh-imports` to refresh the imports (that one is provided by the ODK), `refresh-mappings` to refresh the mappings, `refresh-bridges` to refresh the bridges, and `refresh-external-resources` to refresh everything. Why would `refresh-mappings` be at odds?
`refresh-mappings`, `refresh-external-resources`, `refresh-imports`

Reply:
No worries, I am overthinking it. I was worried the ODK might try to refresh an import called "mappings" when this command is run, due to the refresh-% goal, but this will certainly not be the case. Unless some unwitting ODK user try to configure an import called "mappings" we will probably be good. Ignore!

Reply:
Ah, good point. But no, no risk of any problem here. A rule with an explicit target like `refresh-mappings` will always override a pattern rule like `refresh-%`, so `refresh-mappings` will always do what is expected.

It does mean indeed that Uberon can never define an import called `mappings` (or `bridges`, or `external-resources`), because such imports wouldn’t work. But I can’t imagine Uberon ever having a need to do that. (And even if it were to happen, it wouldn’t be a big deal to rename those targets.)